web/src/popover_menus.js

```diff
@@ -272,6 +288,140 @@ export function initialize() {
         },
     });
 
+    tippy_no_propagation("#stream_filters .topic-sidebar-menu-icon", {
+        ...left_sidebar_tippy_options,
+        onShow(instance) {
+            popover_instances.topics_menu = instance;
+            on_show_prep(instance);
+            const elt = $(instance.reference).closest(".topic-sidebar-menu-icon").expectOne()[0];
```

Comment:
I believe it would be beneficial to extract all the logic related to retrieving data required for rendering into a separate file called popover_menus_data.js. This would improve code readability, leaving only the render logic in the current file. What are your thoughts on this approach?

Reply:
Sounds good, will refactor the code accordingly